src/services/importService.ts

```diff
@@ -0,0 +1,186 @@
+                return assetService.save(assetMetadata);
+            });
+
+            try {
+                Promise.all(saveAssets);
+            } catch (e) {
+                reject(e);
+            }
+
+            resolve(convertedProject);
+        });
+    }
+
+    /**
+     * Generate connections from v1 project file location
+     * @param project - V1 Project Content and File Information
+     */
+    private generateConnections(project: any): IConnection[] {
+        const sourceConnection: IConnection = {
+            id: shortid.generate(),
+            name: "Source Default Name",
+            providerType: "localFileSystemProxy",
+            providerOptions: {
+                folderPath: project.file.path.replace(/[^\/]*$/, ""),
+            },
+        };
+
+        const targetConnection: IConnection = {
+            id: shortid.generate(),
+            name: "Target Default Name",
+            providerType: "localFileSystemProxy",
+            providerOptions: {
+                folderPath: project.file.path.replace(/[^\/]*$/, ""),
+            },
+        };
+
+        const connections: IConnection[] = [sourceConnection, targetConnection];
+
+        return(connections);
+    }
+
+    /**
+     * Parse v1 project's tag string and return array of ITags
+     * @param project - V1 Project Content and File Information
+     */
+    private parseTags(project: any): ITag[] {
+        const finalTags: ITag[] = [];
+        const tagStrings = project.inputTags.split(",");
+        const tagColors = project.tag_colors;
+
+        for (let i = 0; i < tagColors.length; i++) {
+            const newTag = {
+                name: tagStrings[i],
+                color: tagColors[i],
+            };
+            finalTags.push(newTag);
+        }
+        return finalTags;
+    }
+
+    /**
+     * Generate assets based on V1 Project frames and regions
+     * @param project - V1 Project Content and File Information
+     */
+    private generateAssets(project: any): IAssetMetadata[] {
+        let originalProject: IV1Project;
+        let assetMetadata: IAssetMetadata;
+        const generatedAssetMetadata: IAssetMetadata[] = [];
+        let generatedRegion: IRegion;
+        let assetState: AssetState;
+        const currentTagColorIndex = randomIntInRange(0, TagColors.length);
+
+        originalProject = JSON.parse(project.content);
+
+        for (const frameName in originalProject.frames) {
+            if (originalProject.frames.hasOwnProperty(frameName)) {
+                const frameRegions = originalProject.frames[frameName];
+                assetState = originalProject.visitedFrames.indexOf(frameName) > -1 && frameRegions.length > 0
+                             ? AssetState.Tagged : (originalProject.visitedFrames.indexOf(frameName) > -1
+                             ? AssetState.Visited : AssetState.NotVisited);
+
+                assetMetadata = {
+                    asset: {
```

Comment:
We have a method on the `AssetService` to create an asset from a file path.  We should re-use this as much as possible to make sure everything is inline.  Then, once you receive the asset you can then override any specific properties as needed based on the current state of the asset from the v1 project.

Reply:
When I leverage the asset service, it seems to be creating an asset based on the project json file, not the jpg's in that directory, and thus creating only one asset file. Would there be a reason for this?